### PR TITLE
Create mapping-io-extras

### PR DESCRIPTION
**Pull request by modmuss50**
TODO - ProGuardFileWriter depends on ASM.

Java 11+ is required to build this project, Java 8 is still the output target.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -4,12 +4,12 @@ jobs:
   build:
     strategy:
       matrix:
-        java: [8-jdk, 11-jdk, 16-jdk, 17-jdk]
-    runs-on: ubuntu-20.04
+        java: [11-jdk, 16-jdk, 17-jdk, 21-jdk]
```
Comment (NebelNidas) [inline .github/workflows/build.yml:7]: Why Java 16? For MC 1.17? Doesn't the official MC launcher use Java 17 for that version now too?

File: build.gradle [hunk 2]
```diff
@@ -1,180 +1,183 @@
...
+	test {
+		useJUnitPlatform()
 	}
-}
 
-javadoc {
-	failOnError = false
-}
+	javadoc {
+		failOnError = false
```
Comment (NebelNidas) [inline build.gradle:62]: Maybe add https://github.com/FabricMC/fabric-loader/blob/0ba1a7ee50f86a0bbfb2fc7b0a7b33db501211c5/build.gradle#L247-L248 too while you're at it? The log spam is quite annoying when building

**Review by NebelNidas — APPROVED** [review]
Apart from the two minor nitpicks/suggestions above this looks fine to me. Personally I'd prefer if the `extras` module was renamed to `mapping-io-extras` and the main module to `mapping-io-core`, to be better prepared for potential further modularization in the future (and cleaner repository structure), but that's just my opinion.

File: .github/workflows/build.yml (hunk 1 above)
Comment (modmuss50) [reply .github/workflows/build.yml:7]: We can prob remove this, but as far as I'm aware it still uses Java 16

**Comment by modmuss50** [comment]
I don't think we should rename the core package, it gets published as mapping-io-extras. I can rename the directory if that makes more sense?

**Comment by NebelNidas** [comment]
With `modules` I meant Gradle modules and corresponding folder names, so instead of
```
├── extras
│   ├── build.gradle
│   └── src
│       └── ...
├── src
│   └── ...
├── build.gradle
```
I'd prefer
```
├── mapping-io-extras
│   ├── build.gradle
│   └── src
│       └── ...
├── mapping-io-core
│   ├── build.gradle
│   └── src
│       └── ...
├── build.gradle
```

File: .github/workflows/build.yml (hunk 1 above)
Comment (modmuss50) [reply .github/workflows/build.yml]: Removed

**Pull request merged**